config/config.go

```diff
@@ -559,12 +562,12 @@ func validate(c *Config) error {
 	return c.parseForwardedHeaders()
 }
 
-func (c *Config) Parse() error {
-	flag.Parse()
+func (c *Config) Parse(args []string) error {
```

Comment:
This would be API change.

Lets add a new method 
```go
func (c *Config) ParseArgs(name, args []string) error 
```
and call change existing one to the 
```go
func (c *Config) Parse() error {
    return c.ParseArgs(os.Args[0], os.Args[1:])
}
```